### ISSUE TITLE
Something weird is happening with Hammerhead when Polly is trying to capture browser requests

### ISSUE THREAD

**Issue by jordangarside** [DevExpress/testcafe#4372]
Hello,

I'm trying to use [Polly.js](https://github.com/Netflix/pollyjs) to record responses in the browser, while using Testcafe.

```js
// polly/build/index.js
import XHRAdapter from '@pollyjs/adapter-xhr'
import { Polly } from '@pollyjs/core'
import RESTPersister from '@pollyjs/persister-rest'
import FetchAdapter from '@pollyjs/adapter-fetch'

Polly.register(FetchAdapter)
Polly.register(XHRAdapter)
Polly.register(RESTPersister)

window.Polly = Polly
```

```js
// index.test.js
fixture('hello world').clientScripts([
  '../../polly/build/index.js',
  {
    content: `
      window.polly = new Polly('helloWorld', {
        adapters: ['fetch', 'xhr'],
        persister: 'rest',
      })
      window.polly.configure({
        recordFailedRequests: true,
        matchRequestsBy: {
          body: false,
          headers: false,
        },
        persisterOptions: {
          keepUnusedRequests: true,
        },
      })
  `,
  },
])
```

The issue seems to be that not all requests are getting captured by Polly.js, I believe due to the [script getting processed by Hammerhead](https://github.com/DevExpress/testcafe/blob/41b9f22f07c040365f58553b9bbdeb71fbd0f529/src/custom-client-scripts/get-code.ts#L6).

Is there any information as to why this might be happening?

From what I understand the script re-writing is happening so that Testcafe can proxy everything while making the injected scripts think they're still in a normal environment, but I don't understand how some of the requests would get through while others would not.

**Comment by Dmitry-Ostashev** [comment]
Hello,

I suggest you try our built-in [Logging HTTP Requests](https://devexpress.github.io/testcafe/documentation/test-api/intercepting-http-requests/logging-http-requests.html#creating-a-logger) mechanism. This RequestLogger takes into account all the features of request processing (conversion proxy urls into original, image urls, etc.).

Let us know if you need to use exactly the Polly.js tool and our logger does not meet your requirements, we'll try to research how it works with TestCafe.

**Comment by jordangarside** [comment]
Actually found out that this was due to the pollyjs getting reloaded on page changes within a single test (and therefore requests from prior pages would not get included).

Another factor was that every time a new iframe is created, the `clientScripts` that are being used to inject Polly are run, leading it to start recording an iframe's requests.

I've created a demo of using pollyjs with testcafe [here](https://github.com/jordangarside/testcafe-pollyjs-demo).

Now I just need to figure out [this issue](https://github.com/DevExpress/testcafe/issues/4376).
